### PR TITLE
Tic tac toe finished

### PR DESCRIPTION
**Pull request by bouaik**
## Multiplayer tic tac toe 
- players can put their names
- players can create a new game after finishing a game

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Tic tac toe
+
+> This is a Tic Tac game built by javascript
+
+![captured](https://user-images.githubusercontent.com/45256093/92619084-f960db00-f2b8-11ea-9f39-e4e3e2e82003.gif)
+
+
+
+# Features
+- Both players can put thier names
```
Comment (mosaaleb) [inline README.md:10]: - [ ] Typo in `thier`

File: app.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  this.symbol = symbol;
+}
+
+const playerOne = new Player('', 'X');
+const playerTwo = new Player('', 'O');
+
+
+const currentPlayerTurn = (cp) => `It's ${cp}'s turn`;
+
+let gameState = ['', '', '', '', '', '', '', '', ''];
```
Comment (mosaaleb) [inline app.js]: - [ ] **[OPTIONAL]** We can use array class initialization to have more idiomatic, less repititive code: 

```js
let gameState = new Array(9).fill('');

File: app.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+const form = document.getElementById('form');
+const alert = document.querySelector('.alert');
+let gameActive = true;
+
+function Player(name, symbol) {
+  this.name = name;
+  this.symbol = symbol;
+}
+
+const playerOne = new Player('', 'X');
```
Comment (mosaaleb) [inline app.js]: - [ ] It doesn't use global code - encapsulate logic inside modules or factories

File: app.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
+const statusDisplay = document.querySelector('.game--status');
```
Comment (mosaaleb) [inline app.js]: - [ ] It doesn't use global code. Let's encapsulate all DOM functionalities inside its own module, and expose its API function for other modules to use. Creating a module for the DOM functionality is also very important for separating code logic from dom manipulation. See next point
 
- [ ] Keep the application logic separated from DOM manipulation tasks


A possible solution is to create a `DOM` module which should contain the DOM access and manipulation and a `render` function whose only responsibility is to draw the board on screen from the board array.

File: app.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+
+const currentPlayerTurn = (cp) => `It's ${cp}'s turn`;
+
+let gameState = ['', '', '', '', '', '', '', '', ''];
+
+/* eslint no-use-before-define: ["error", { "functions": false }] */
+
+function handleCellClick(clickedCellEvent) {
+  const clickedCell = clickedCellEvent.target;
```
Comment (mosaaleb) [inline app.js]: - [ ] It has a function with the only responsibility of rendering the contents of the gameboard array to the webpage. 



handleCellClick is doing too much, it listens for cell click, and then with the help of `handleCellPlayed` will update the board array, and then updating the dom (game logic is not separated from dom manipulation)

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Well done @bouaik :clap:

The gameplay is very nice and doesn't have any bugs. Very well done. Yet there are other requirements related to code organization that needs to be met before moving on:

- [ ] The goal of this project is to make use of modules and factory functions and expose specific API functions for other modules to interact with it. Your code is organized but we need to use modules and factory functions patterns to get this project approved.

- [ ] It has game, gameBoard, player objects



You have only created a player object using factory functions. But we need to use the module pattern to encapsulate functionality for `Game` and `GameBoard`. Again, you can use module pattern if it's only one thing you are creating i.e. (Game, GameBoard), and factory function pattern -- like you did -- with objects that can be created more than once.


**To Summarize the requirements**:

- You need to use the module pattern for creating `Game` and `GameBoard`. You can start by asking yourself some questions about each one of these and what methods each module should have. For example, `Game` should have a method to check if we have reached an endgame position. `GameBoard` should have a method to check if a cell is occupied or not, the array board should be defined in `GameBoard`. Players also could be initialized inside the `Game` module. That way everything is encapsulated and our code are more organized.

Please handle these requirements and submit a new code review.
Feel free to contact me on Slack if you have any questions

##### Muhammad Ebeid

**Comment by bouaik** [comment]
## Changes Implemented
- create factory function for Board, Game, Player
- separate DOM from game Logic

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @bouaik 
### Good job you have done so far :clap: However, you have a few issues you need to fix:-
- [ ] Separating dom from app logic means having two files for logic and dom
- [ ] Update your readme on how one can install the project and play the game
### Make the changes and submit for another code review. Good luck
- Happy coding!

### [Optional]
- You can use localStorage to store users scores

By Raymond
Slack @iamraymond

File: app.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    currentPlayerTurn,
+    winningMessage,
+    drawMessage,
+    showBoard,
+    statusDisplay,
+    showAlert,
+    removeBoard,
+    clearFields,
+  };
+})();
```
Comment (rOluochKe) [inline app.js]: ### Take the dom in a separate file

**Review by rOluochKe — APPROVED** [review]
### You got it right. The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

### [Optional]
- Create a directory, src and put your development files inside for easy readability and maintenance
